linuxrc_hostname.md

```diff
@@ -1,30 +1,35 @@
 # hostname setting in installation/rescue system
 
-There are 3 cases:
+The hostname can be set (in order of precedence):
 
-1. no hostname is set
+1. via `hostname` boot option
+2. via DHCP (either from HOSTNAME or FQDN)
+3. Default hostname `install`
 
-2. hostname is set via dhcp (either from HOSTNAME or FQDN)
+linuxrc defaults to `install`, which may be overridden by DHCP(2) or user(1).
+(1) takes precedence over (2).
 
-3. hostname is set via `hostname` boot option
-
-linuxrc defaults to `install`, which may be overridden by dhcp (2) or user
-(3). (3) takes precedence over (2).
 
 ## installation system
 
-When yast is started, the value from (3) is passed to yast via
-`/etc/install.inf::Hostname` and ends up in `/etc/hostname` of the installed system.
+When yast is started, the value from (1) is passed to yast via
+`/etc/install.inf::Hostname` and ends up in `/etc/hostname` of the installed
+system.
+
+When linuxrc doesn't set `/etc/install.inf::Hostname`, yast generates a default
+`/etc/hostname` entry:
+- Before SLE12, of the form `linux-XXXX`
+- Since SLE12, `linux-XXXX.suse`
+- Since SLE15-SP2, `linux`
+`XXXX` is a random-generated alphanumerical string.
 
-When linuxrc doesn't set `/etc/install.inf::Hostname` yast generates a
-default `/etc/hostname` entry of the form `linux-XXXX` (+ `.suse` in sle12).
-`XXXX` is some random part.
+linuxrc doesn't generate the `/etc/hostname` file in the installation
```

Comment:
Well, it is not a generation at all. When a `hostname` is configured through linuxrc, yast writes the given hostname to /etc/hostname at very beginning of the installation.

https://github.com/yast/skelcd-control-leanos/blob/master/control/control.leanos.xml#L516
https://github.com/yast/yast-installation/blob/master/src/lib/installation/clients/inst_install_inf.rb#L19
https://github.com/yast/yast-network/blob/c1ed7b6ca13a391b35a7de95458bcc628da26084/src/lib/network/install_inf_convertor.rb#L61

Reply:
> YaST writes the given hostname to /etc/hostname at very beginning of the installation

I don't see that happen in my test. /etc/hostname **in the installation system** is always empty.

Reply:
At which point are you checking it?

Reply:
In which release?, and have you checked once YaST starts?

Reply:
As described, it is set by the very first client of the installation workflow.

![Screenshot_sles12sp3_2020-04-06_09:08:39](https://user-images.githubusercontent.com/7056681/78536816-562a0800-77e6-11ea-8506-e75ec8ba3f8c.png)
![Screenshot_sles12sp3_2020-04-06_09:10:04](https://user-images.githubusercontent.com/7056681/78536877-6d68f580-77e6-11ea-9d09-1e76301b9ffa.png)

Reply:
And the same in SLE-15:

![Screenshot_sle15_2020-04-06_09:48:29](https://user-images.githubusercontent.com/7056681/78540279-d737ce00-77eb-11ea-99fb-59360c4afbc7.png)

Reply:
Strange, but you are correct. No idea what I was looking at.